src/ui/store/slice/shareData/shareData.slice.ts

```diff
@@ -97,85 +307,122 @@ const initFormInvariant =
       ),
-    setFormItemValue: ({ id, value }: SetFormItemValuePayload) =>
-      pipe(
-        id,
-        S.isEmpty,
-        B.match(
-          () =>
-            pipe(
-              IOE.fromIO(() => get().shareData.form),
-              IOE.flatMap(
-                flow(
-                  setFormItemValueInvariant(id),
-                  IOE.fromEither,
-                  IOE.chainIOK(
-                    formItemId => () =>
-                      set(state => ({
-                        shareData: {
-                          ...state.shareData,
-                          form: pipe(
-                            state.shareData.form,
-                            A.map(formItem =>
-                              pipe(
-                                eqFormItem.equals(formItem, { id: formItemId }),
-                                B.match(
-                                  () => formItem,
-                                  () => ({ ...formItem, value })
+      initForm: (payload: InitFormPayload) =>
+        pipe(
+          payload,
+          A.isEmpty,
+          B.matchW(
+            () =>
+              pipe(
+                IOE.fromIO(() => get().shareData.form),
+                IOE.flatMap(
+                  flow(
+                    initFormInvariant(payload),
+                    IOE.fromEither,
+                    IOE.chainIOK(
+                      form => () =>
+                        set(state => ({
+                          shareData: {
+                            ...state.shareData,
+                            form
+                          }
+                        }))
+                    )
+                  )
+                )
+              ),
+            () => IOE.left(emptyPayloadError(payload))
+          )
+        ),
+      formItemById: (id: FormItemId) =>
+        pipe(
+          IOO.fromIO(() => get().shareData.form),
+          IOO.chainOptionK(flow(A.findFirst(formItem => eqFormItem.equals(formItem, { id }))))
+        ),
+      setFormItemValue: (id: string, value: number | string | I18nStringPayload) =>
+        pipe(
+          id,
+          S.isEmpty,
+          B.matchW(
+            () =>
+              pipe(
+                IOE.fromIO(() => get().shareData.form),
+                IOE.flatMap(
+                  flow(
+                    setFormItemValueInvariant(id),
+                    IOE.fromEither,
+                    IOE.chainW(
+                      flow(
+                        updateFormItem(value),
+                        IOE.fromEither,
+                        IOE.chainIOK(
+                          formItem => () =>
+                            set(state => ({
+                              shareData: {
+                                ...state.shareData,
+                                form: pipe(
+                                  state.shareData.form,
+                                  A.map(it => (S.Eq.equals(it.id, id) ? formItem : it))
```

Comment:
From what I see, the code iterates over each element in the array, comparing the id of each item with a specific id. If a match is found, the item is replaced with `formItem`; if not, the item remains unchanged.

However, this is not the primary purpose of a `map` function. For both performance and clarity, it would be more appropriate to first locate the item and then perform the change.

Moreover if such operations are common throughout the codebase, it might be wise to put them into a utility function that combines a find operation with a map operation (making the operation more explicit).